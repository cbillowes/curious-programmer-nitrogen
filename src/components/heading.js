import React from 'react'
import styled from 'styled-components'
import Constants from '../../gatsby-data'

const Heading = styled.div`
  font-family: ${Constants.theme.fonts.sans};
  font-size: 130%;
  line-height: 2.75rem;
`

const H1 = ({ className, children }) => {
  return (
    <Heading className={className}><h1>{children}</h1></Heading>
  )
}

const H2 = ({ className, children }) => {
  return (
    <Heading className={className}><h2>{children}</h2></Heading>
  )
}

const H3 = ({ className, children }) => {
  return (
    <Heading className={className}><h3>{children}</h3></Heading>
  )
}

const H4 = ({ className, children }) => {
  return (
    <Heading className={className}><h4>{children}</h4></Heading>
  )
}

export { H1, H2, H3, H4 }
