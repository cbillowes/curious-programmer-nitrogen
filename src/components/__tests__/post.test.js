import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { containsElement } from './helpers'
import Post from '../post'
import { H1 } from '../heading'
import Anchor from '../anchor'
import PostMetadata from '../postMetadata'
import Tags from '../tags'
import Constants from '../../../gatsby-data'

const colors = Constants.theme.colors;

Enzyme.configure({ adapter: new Adapter() })

function sut(props) {
  const { date, author, ttr } = props.metadata || {}
  return (
    <Post 
      title={props.title}
      slug={props.slug}
      tags={props.tags}
      date={date}
      author={author}
      ttr={ttr}
    >
      {props.children}
    </Post>
  )
}

describe('Post', () => {

  it('should render the title', () => {
    const contains = containsElement(
      sut({
        title: `Hello world`,
        slug: `/blog/hello-world`,
      }),
      <H1>
        <Anchor
          to="/blog/hello-world"
          title="Hello world"
          bland={true}
          style={{
            color: `${colors.light}`,
            textDecoration: `none`,
            borderBottom: `solid 1px ${colors.light}`,
          }}
        >
          Hello world
        </Anchor>
      </H1>
    )
    expect(contains).toBe(true)
  })

  it('should render post metadata', () => {
    const contains = containsElement(
      sut({
        slug: `/`,
        metadata: {
          date: `1 April 2019`,
          author: `Douglas Adams`,
          ttr: `15`,
        }
      }),
      <PostMetadata 
        date="1 April 2019"
        author="Douglas Adams"
        ttr="15"
      />
    )
    expect(contains).toBe(true)
  })

  it('should render the tags', () => {
    const contains = containsElement(
      sut({
        slug: `/`,
        tags: `hello,world,greetings` 
      }),
      <Tags
        tags={`hello,world,greetings`}
      />
    )
    expect(contains).toBe(true)
  })

  it('should render the body', () => {
    const contains = containsElement(
      sut({
        slug: `/`,
        children: `Lorem ipsum.`,
      }),
      <div>
        Lorem ipsum.
      </div>
    )
    expect(contains).toBe(true)
  })

})
