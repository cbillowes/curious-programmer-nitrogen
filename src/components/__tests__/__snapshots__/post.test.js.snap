// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render with a number 1`] = `
<article
  className="post article"
>
  <div
    className="post number ribbon"
  >
    <div
      className="backdrop"
    >
      <div
        className="text"
      >
        #
        42
      </div>
    </div>
  </div>
  <h1>
    Don't panic.
  </h1>
  <div
    className="metadata"
  >
    posted on
     
    <time
      dateTime={"2019-02-01T00:00:00.000Z"}
    >
      Friday, 1 February 2019
    </time>
     
    <span>
      by Clarice Bouwer
    </span>
     
    <span>
      (Est. 42 minute read)
    </span>
  </div>
  <div
    className="tags"
  >
     
    <a
      className="tag"
      href="/tag/dont/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="don't"
    >
      don't
    </a>
     
     
    <a
      className="tag"
      href="/tag/panic/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="panic"
    >
      panic
    </a>
     
  </div>
  <div
    className="reading-panel"
  >
    <p>
      Don't panic.
    </p>
  </div>
</article>
`;

exports[`Post should render with html 1`] = `
<article
  className="post article"
>
  <h1>
    Don't panic.
  </h1>
  <div
    className="metadata"
  >
    posted on
     
    <time
      dateTime={"2019-02-01T00:00:00.000Z"}
    >
      Friday, 1 February 2019
    </time>
     
    <span>
      by Clarice Bouwer
    </span>
     
    <span>
      (Est. 42 minute read)
    </span>
  </div>
  <div
    className="tags"
  >
     
    <a
      className="tag"
      href="/tag/dont/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="don't"
    >
      don't
    </a>
     
     
    <a
      className="tag"
      href="/tag/panic/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="panic"
    >
      panic
    </a>
     
  </div>
  <div
    className="reading-panel"
  >
    <p>
      Arthur Dent (Martin Freeman) is trying to prevent his house from being bulldozed when his friend Ford Prefect (Mos Def) whisks him into outer space. It turns out Ford is an alien who has just saved Arthur from Earth's total annihilation. Ford introduces Arthur to his myriad friends, including many-headed President Zaphod Beeblebrox (Sam Rockwell) and sexy refugee Trillian (Zooey Deschanel). Arthur makes his way across the stars while seeking the meaning of life, or something close to it.
    </p>
  </div>
</article>
`;

exports[`Post should render with plain text 1`] = `
<article
  className="post article"
>
  <h1>
    Don't panic.
  </h1>
  <div
    className="metadata"
  >
    posted on
     
    <time
      dateTime={"2019-02-01T00:00:00.000Z"}
    >
      Friday, 1 February 2019
    </time>
     
    <span>
      by Clarice Bouwer
    </span>
     
    <span>
      (Est. 42 minute read)
    </span>
  </div>
  <div
    className="tags"
  >
     
    <a
      className="tag"
      href="/tag/dont/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="don't"
    >
      don't
    </a>
     
     
    <a
      className="tag"
      href="/tag/panic/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="panic"
    >
      panic
    </a>
     
  </div>
  <div
    className="reading-panel"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Arthur Dent (Martin Freeman) is trying to prevent his house from being bulldozed when his friend Ford Prefect (Mos Def) whisks him into outer space. It turns out Ford is an alien who has just saved Arthur from Earth's total annihilation. Ford introduces Arthur to his myriad friends, including many-headed President Zaphod Beeblebrox (Sam Rockwell) and sexy refugee Trillian (Zooey Deschanel). Arthur makes his way across the stars while seeking the meaning of life, or something close to it.",
      }
    }
  />
</article>
`;
